src/transceivercontroller/DefaultTransceiverController.ts

```diff
@@ -402,4 +442,366 @@ export default class DefaultTransceiverController implements TransceiverControll
 
     await transceiver.sender.replaceTrack(track);
   }
+
+  private forEachRedMetricsObserver(
+    observerFunc: (observer: RedundantAudioRecoveryMetricsObserver) => void
+  ): void {
+    for (const observer of this.redMetricsObservers) {
+      AsyncScheduler.nextTick(() => {
+        /* istanbul ignore else */
+        // Since this executes asynchronously, we need to check if the observer has not been removed.
+        if (this.redMetricsObservers.has(observer)) {
+          observerFunc(observer);
+        }
+      });
+    }
+  }
+
+  private setupAudioRedWorker(): void {
+    // @ts-ignore
+    const supportsRTCScriptTransform = !!window.RTCRtpScriptTransform;
+    // @ts-ignore
+    const supportsInsertableStreams = !!RTCRtpSender.prototype.createEncodedStreams;
+
+    if (supportsRTCScriptTransform) {
+      // This is the prefered approach according to
+      // https://github.com/w3c/webrtc-encoded-transform/blob/main/explainer.md
+      // but chrome doesn't seem to support it as of chrome 111.0
+      // Safari 16.3 seems to support this.
+      this.logger.info(
+        '[AudioRed] Supports encoded insertable streams using RTCRtpScriptTransform'
+      );
+    } else if (supportsInsertableStreams) {
+      this.logger.info('[AudioRed] Supports encoded insertable streams using TransformStream');
+    } else {
+      this.logger.warn('[AudioRed] Encoded insertable streams not supported');
+      return;
+    }
+
+    // Stringify the `RedundantAudioEncoder` class code and get the new name of the `RedundantAudioEncoder` class since
+    // its name is changed in the browser.
+    const redWorkerCode = RedundantAudioEncoder.toString();
+    const redClassName = redWorkerCode.match(/class\s*(\w+)\s*\{/)[1];
+
+    const blobParts: BlobPart[] = [redWorkerCode];
+    if (this.logger.getLogLevel() === LogLevel.DEBUG) {
+      blobParts.push(`${redClassName}.shouldLogDebug=1;`);
+    }
+    blobParts.push(`${redClassName}.shouldReportStats=1;`);
+    blobParts.push(`${redClassName}.initializeWorker();`);
+
+    this.audioRedWorkerURL = URL.createObjectURL(
+      new Blob(blobParts, {
+        type: 'application/javascript',
+      })
+    );
+    this.logger.info(`[AudioRed] Redundant audio worker URL ${this.audioRedWorkerURL}`);
+    this.audioRedWorker = new Worker(this.audioRedWorkerURL);
```

Comment:
Can this fail? Should we catch error here?

Reply:
I guess it can theoretically fail. Lemme force the `audioRedWorker` to `null` and see behavior.

Reply:
Looks like this could fail for reasons mentioned in https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker

Updated to account for failure and restart connection with redundancy disabled so that user connects to the meeting. Tested this out by removing `blob` from csp worker-src and connecting to a meeting. Tested both audio and video sending and receiving. Also added unit tests.